homeassistant/components/diagnostics/__init__.py

```diff
@@ -121,17 +123,44 @@ def handle_get(
     )
 
 
-def _get_json_file_response(
+async def _async_get_json_file_response(
+    hass: HomeAssistant,
     data: dict | list,
     filename: str,
+    domain: str,
     d_type: DiagnosticsType,
     d_id: str,
     sub_type: DiagnosticsSubType | None = None,
     sub_id: str | None = None,
 ) -> web.Response:
     """Return JSON file from dictionary."""
+    hass_sys_info = await async_get_system_info(hass)
+
+    integration: Integration = hass.data[DATA_INTEGRATIONS][domain]
+
+    custom_components = {}
+    cc_domain: str
+    cc_obj: Integration
+    for cc_domain, cc_obj in hass.data[DATA_CUSTOM_COMPONENTS].items():
+        custom_components[cc_domain] = {
+            "version": cc_obj.manifest["version"],
+            "requirements": [
+                req
+                for req in cc_obj.manifest["requirements"]
+                if "://" not in req  # ignore url based requirements
+            ],
+        }
     try:
-        json_data = json.dumps(data, indent=2, cls=ExtendedJSONEncoder)
+        json_data = json.dumps(
+            {
+                "home-assistant": hass_sys_info,
+                "custom-components": custom_components,
+                "integration-manifest": integration.manifest,
```

Comment:
Could be me, but my OCD brain is going on this one...

```suggestion
                "home_assistant": hass_sys_info,
                "custom_components": custom_components,
                "integration_manifest": integration.manifest,
```